src/TCommandLine.cpp

```diff
@@ -566,56 +578,181 @@ void TCommandLine::slot_popupMenu()
     c.clearSelection();
-    Hunspell_free_list(mpHunspell, &mpHunspellSuggestionList, mHunspellSuggestionNumber);
+    Hunspell_free_list(mpHost->mpConsole->getHunspellHandle_system(), &mpHunspellSystemSuggestionsList, mHunspellSystemDictionarySuggestionsCount);
+    Hunhandle* userDictionaryHandle = mpHost->mpConsole->getHunspellHandle_user();
+    if (userDictionaryHandle) {
+        Hunspell_free_list(userDictionaryHandle, &mpHunspellProfileSuggestionsList, mHunspellProfileDictionarySuggestionsCount);
+    }
 }
 
 void TCommandLine::mousePressEvent(QMouseEvent* event)
 {
 
     if (event->button() == Qt::RightButton) {
         auto popup = createStandardContextMenu(event->globalPos());
-        QTextCursor c = cursorForPosition(event->pos());
-        c.select(QTextCursor::WordUnderCursor);
-        QByteArray encodedText = mpHunspellCodec->fromUnicode(c.selectedText());
-        if (!Hunspell_spell(mpHunspell, encodedText.constData())) {
-            // The word is NOT in the dictionary:
-            char** sl;
-            auto separator = popup->actions().first();
-            separator = popup->insertSeparator(separator);
-            // separator is now the QAction of, indeed is, a separator:
-            QList<QAction*> spellings;
+        if (mpHost->mEnableSpellCheck) {
+            QTextCursor c = cursorForPosition(event->pos());
+            c.select(QTextCursor::WordUnderCursor);
+            mSpellCheckedWord = c.selectedText();
+            auto codec = mpHost->mpConsole->getHunspellCodec_system();
+            auto handle_system = mpHost->mpConsole->getHunspellHandle_system();
+            auto handle_profile = mpHost->mpConsole->getHunspellHandle_user();
+            QByteArray encodedText = codec->fromUnicode(mSpellCheckedWord);
+            bool haveAddOption = false;
+            bool haveRemoveOption = false;
+            QAction* action_addWord = nullptr;
+            QAction* action_removeWord = nullptr;
+            QAction* action_dictionarySeparatorLine = nullptr;
+            if (handle_profile) {
+                // TODO: Make icons for these?
+//                if (!qApp->testAttribute(Qt::AA_DontShowIconsInMenus)) {
+//                    action_addWord = new QAction(QIcon(QPixmap(QStringLiteral(":/icons/dictionary-add-word.png"))), tr("Add to user dictionary"));
+//                    action_removeWord = new QAction(QIcon(QPixmap(QStringLiteral(":/icons/dictionary-remove-word.png"))), tr("Remove from user dictionary"));
+//                } else {
+                action_addWord = new QAction(tr("Add to user dictionary"));
+                action_addWord->setEnabled(false);
+                action_removeWord = new QAction(tr("Remove from user dictionary"));
+                action_removeWord->setEnabled(false);
+//                }
+                if (mudlet::self()->mUsingMudletDictionaries) {
+                    action_dictionarySeparatorLine = new QAction(tr("▼Mudlet▼ │ dictionary suggestions │ ▲User▲",
```

Comment:
As before - some users (like myself) will want to know if it is a word from the MUD or added.

Perhaps it is less of an issue when such a word has been added by hand, but, for instance I would anticipate having triggers to add the names of all the players I identify in MUDs I play to that profile's dictionary so I always have them stored even if I haven't encountered them this session (so they are not in the tab completion list).  As there is considerable scope for automation via scripts in this (I think) - I do not think it is unreasonable to keep the two sets of words distinct - in fact - they have to be because they come from two completely unrelated sets of Hunspell files/dictionaries and merging them would actually take more code I believe.  Of course I could remove the decoration permanently (as happens if the user one is disabled) so there would seem to be just one list of suggestions - it is just that with the user one enabled the list could be up to twice the size as each is subject to a compiled in Hunspell limit.

I wonder if the @Mudlet/ui-review team have any input?